File: nomenclator/src/main/java/flowmanager/nomenclator/model/Organization.java
package flowmanager.nomenclator.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(exclude = {
        "manager",
        "members",
        "teams",
        "projects"
})
@Builder
@Entity
@Table(name = "organization")
public class Organization {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private String industry;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "manager_id", nullable = false)
    private User manager;

    @Builder.Default
    @ManyToMany
    @JoinTable(name = "organization_user",
            joinColumns = @JoinColumn(name = "organization_id"),
            inverseJoinColumns = @JoinColumn(name = "user_id"))
    private List<User> members = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "organization")
    private List<Team> teams = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "organization")
    private List<Project> projects = new ArrayList<>();

}

File: frontend/app/types/organization.ts
import type { UserSummaryDto } from "./user"

export interface OrganizationCreateDto {
  name: string
  description: string
  industry: string
  managerId: number
}

export interface OrganizationUpdateDto {
  name: string
  description: string
  industry: string
  managerId: number | null
}

export interface OrganizationSummaryDto {
  id: number
  name: string
  description: string
}

export interface OrganizationResponseDto {
  id: number
  name: string
  description: string
  industry: string
  createdAt: string
  manager: UserSummaryDto
  teamCount: number
  projectCount: number
  memberCount: number
}